File: templates/security/api/src/Infrastructure/Security/Crypto/CtxProtocol.cs
namespace CtxApp.Infrastructure.Security.Crypto;

/// <summary>
/// Constants that define the ctx.0 wire protocol. The Flutter client and this
/// API implement the same values; both test suites assert against the shared
/// golden vectors in <c>protocol/vectors.json</c>.
/// </summary>
public static class CtxProtocol
{
    /// <summary>Protocol version advertised in the <c>X-Ctx-Protocol</c> header.</summary>
    public const string Version = "1.0";

    // Request headers.
    public const string ProtocolHeader = "X-Ctx-Protocol";
    public const string DeviceIdHeader = "X-Ctx-Device-Id";
    public const string TimestampHeader = "X-Ctx-Timestamp";
    public const string SignatureHeader = "X-Ctx-Signature";

    /// <summary>HKDF <c>info</c> string binding derived keys to this scheme/version.</summary>
    public const string AleHkdfInfo = "ctx-ale-v1";

    /// <summary>Maximum accepted clock skew for a signed request.</summary>
    public static readonly TimeSpan SignatureWindow = TimeSpan.FromMinutes(5);
}

File: test1/acme/app/lib/security/crypto/ctx_protocol.dart
/// Constants that define the ctx.0 wire protocol on the client. These mirror the
/// API's `CtxProtocol` exactly; both test suites assert against the shared golden
/// vectors in the workspace's `.ctx/vectors.json`.
class CtxProtocol {
  const CtxProtocol._();

  /// Protocol version advertised in the `X-Ctx-Protocol` header.
  static const String version = '1.0';

  static const String protocolHeader = 'X-Ctx-Protocol';
  static const String deviceIdHeader = 'X-Ctx-Device-Id';
  static const String timestampHeader = 'X-Ctx-Timestamp';
  static const String signatureHeader = 'X-Ctx-Signature';

  /// HKDF `info` string binding derived keys to this scheme/version.
  static const String aleHkdfInfo = 'ctx-ale-v1';
}
